io-engine/src/lvs/lvs_lvol.rs

```diff
@@ -602,23 +613,16 @@ impl Lvol {
             .and_then(|b| Lvol::try_from(b).ok())
             {
                 Some(parent) => (parent.uuid(), parent.size()),
-                None => {
-                    warn!("Snapshot{:?}, parent not proper", self);
-                    // valid_snapshot = false;
-                    (String::default(), 0)
-                }
+                None => (String::default(), 0),
```

Comment:
When can this happen, this we see this message in control-plane tests, this is not expected?

Reply:
It is expected when the replica is deleted and We are trying to list the Snapshot, then We show the parent information with default value.

Reply:
hmm in control-plane the replica is not deleted IIRC, we're doing a list all snapshots

Reply:
> hmm in control-plane the replica is not deleted IIRC, we're doing a list all snapshots

Currently control-plane is calling list all snapshots without any attribute. The issue what you have faced last time, because of wrong parent-id details saved for NexusSnapshot. 
Regarding Dataplane interface point of view, We support 3 different kind of ListSnapshot
1. List All Snapshot (Currently in use by CP)
2. List Snapshot based on Source_uuid
3. List Snapshot based on snapshot_uuid

Reply:
We also use 3.
Oh I see, the wrong parent id caused the list to think it didn't have the parent, got it, thanks for explaining :+1: